docs/architecture/adt-014-proportional-slashing.md

```diff
@@ -0,0 +1,78 @@
+
+We can achieve this by not only taking into account the sum of the percentages of the validators that faulted, but also the *number* of validators that faulted in the window.  One general form for an equation that fits this desired property looks like this:
+
+```
+slash_amount = k * (sqrt(power_1) + sqrt(power_2) + ... + sqrt(power_n))^2
+```
+
+So now, for example, (assuming k = 1) if one validator of 10% faults, it gets a 10% slash, while if two validators of 5% each fault together, they both get a 20% slash ((sqrt(0.05)+sqrt(0.05))^2).
+
+The k constant can be provided by the governable parameters, to for example, punish double signs more heavily than liveness faults.  The k factor can also be something other than a constant, there is some research on using things like inverse gini coefficients to mitigate some griefing attacks, but this an area for future research.
+
+One will note, that this model doesn't differentiate between multiple validators run by the same operators vs validators run by different operators.  This can be seen as an additional benefit in fact.  It incentivizes validators to differentiate their setups from other validators, to avoid having correlated faults with them or else they risk a higher slash.  So for example, operators should avoid using the same popular cloud hosting platforms or using the same Staking as a Service providers.  This will lead to a more resilient and decentralized network.
+
+### Implementation
+
+In the slashing module, we will add two queues that will track all of the recent slash events.  For double sign faults, we will define "recent slashes" as ones that have occured within the last `unbonding period`.  For liveness faults, we will define "recent slashes" as ones that have occured withing the last `jail period`.
+
+```
+type SlashEvent struct {
+    Address                 sdk.ValAddress
+    ValidatorVotingPercent  sdk.Dec
+    SlashedSoFar            sdk.Dec
+}
+```
+
+These slash events will be pruned from the queue once they are older than their respective "recent slash period".
+
+Whenever a new slash occurs, a `SlashEvent` struct is created with the faulting validator's voting percent and a `SlashedSoFar` of 0.  Because recent slash events are pruned before the unbonding period and unjail period expires, it should not be possible for the same validator to have multiple SlashEvents in the same Queue at the same time.
```

Comment:
Is it possible for a validator to be in both queues? Does that introduce any issues?

Reply:
Yes, a validator can be in both queues.  I don't think it should cause issues, because the `SquareOfSumOfRoots` calculation is independent for different types of faults.  I don't count for correlation of double signs with liveness.